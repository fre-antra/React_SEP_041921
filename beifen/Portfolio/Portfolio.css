.projects {
  background-color: #59585d;
  padding: 50px;
  color: white;
}

.projects-header-quote {
  margin-bottom: 20px;
}
.projects h1,
.projects h3 {
  font-family: "Source Sans Pro", Arial, sans-serif;
}
.projects p {
  font-family: "Roboto", Helvetica, serif;
  padding: 0 10px;
}

.projects-header-quote {
  font-style: italic;
}
/*.luvtalk-project, .personal-website-project, .strike-zone-project, .resume {border: 3px solid white;}*/
.luvtalk-project {
  background: linear-gradient(rgba(0, 0, 0, 0.85), rgba(0, 0, 0, 0.85)),
    url(../Images/luvtalklogo.png);
  background-position: center center;
}

.personal-website-project {
  background: linear-gradient(rgba(0, 0, 0, 0.85), rgba(0, 0, 0, 0.85)),
    url(../Images/logo.svg);
  background-position: center center;
}

.strike-zone-project {
  background: linear-gradient(rgba(0, 0, 0, 0.85), rgba(0, 0, 0, 0.85)),
    url(../Images/mlbproject.png);
  background-position: center center;
}

.resume {
  padding: 50px 0px;
  font-family: "Calisto MT", Helvetica, serif;
}

.resume-header {
  margin-bottom: 20px;
}
.objective h4,
.education h4,
.work-experience h4,
.resume-projects h4,
.activities h4,
.skills h4,
.endorsements h4 {
  border-bottom: 1px solid black;
}
