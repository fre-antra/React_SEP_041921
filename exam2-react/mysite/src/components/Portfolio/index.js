import { useEffect } from "react";
import $ from "jquery";
const Portfolio = () => {
  useEffect(() => {
    projects();
  }, []);

  function projects() {
    $(".luvtalk-project").css("height", $(".luvtalk-project").width());
    $(".luvtalk-project h3").css(
      "padding-top",
      ($(".luvtalk-project").height() -
        $(".luvtalk-project h3").height() -
        $(".luvtalk-project .row").outerHeight()) /
        2
    );
    $(".personal-website-project").css(
      "height",
      $(".personal-website-project").width()
    );
    $(".personal-website-project h3").css(
      "padding-top",
      ($(".personal-website-project").height() -
        $(".personal-website-project h3").height() -
        $(".personal-website-project .row").outerHeight()) /
        2
    );
    $(".strike-zone-project").css("height", $(".strike-zone-project").width());
    $(".strike-zone-project h3").css(
      "padding-top",
      ($(".strike-zone-project").height() -
        $(".strike-zone-project h3").height() -
        $(".strike-zone-project .row").outerHeight()) /
        2
    );

    var doit;
    $(window).resize(function () {
      clearTimeout(doit);
      doit = setTimeout(projects, 500);
    });
  }

  return (
    <div className="projects" id="#projects">
      <div className="container-fluid">
        <div className="row">
          <div className="col-sm-12 col-md-12 text-center">
            <div className="projects-header">
              <h1>Projects and Portfolio</h1>
            </div>
          </div>
        </div>

        <div className="row">
          <div className="col-sm-12 col-md-12 text-center">
            <div className="projects-header-quote">
              <h3>Sharing my endeavors and passions...</h3>
            </div>
          </div>
        </div>

        <div className="row">
          <div className="col-sm-4 col-md-4 text-center">
            <div className="luvtalk-project">
              <h3>LUV TALK Website</h3>
              <p>
                Website, created during my first internship, which utilized
                HTML, CSS, PHP, Ajax, Javascript/jQuery, and Wordpress.{" "}
              </p>
              <div className="row">
                <div className="col-sm-4 col-md-4 col-sm-offset-4 col-md-offset-4">
                  <div className="project-more-info">
                    <a className="btn btn-default btn-border" href="#">
                      More
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>

          <div className="col-sm-4 col-md-4 text-center">
            <div className="personal-website-project">
              <h3>Personal Website</h3>
              <p>
                Enjoyable side project that was created to experiment with more
                HTML and CSS, but also provided an outlet to showcase my
                abilities and interests.
              </p>
              <div className="row">
                <div className="col-sm-4 col-md-4 col-sm-offset-4 col-md-offset-4">
                  <div className="project-more-info">
                    <a className="btn btn-default btn-border" href="#">
                      More
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>

          <div className="col-sm-4 col-md-4 text-center">
            <div className="strike-zone-project">
              <h3>Strike Zone Analysis</h3>
              <p>
                Data analytics project completed during my time at the Illinois
                Math and Science Academy which studied the baseball strike zone
                based on the state of the game.
              </p>
              <div className="row">
                <div className="col-sm-4 col-md-4 col-sm-offset-4 col-md-offset-4">
                  <div className="project-more-info">
                    <a className="btn btn-default btn-border" href="#">
                      More
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>

        <div className="row">
          <div className="col-sm-2 col-md-2 col-sm-offset-5 col-md-offset-5">
            <div className="see-more-button">
              <a className="btn btn-default btn-border" href="#">
                More Projects
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Portfolio;
