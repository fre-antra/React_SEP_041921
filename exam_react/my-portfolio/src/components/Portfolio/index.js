import React, { Component, Fragment } from 'react'

export default class Portfolio extends Component {
    render() {
        return (
            <div class="projects" id="projects">
                <div class="container-fluid">
                    <div class="row">
                        <div class="col-sm-12 col-md-12 text-center">
                            <div class="projects-header">
                                <h1>Projects and Portfolio</h1>
                            </div>
                        </div>
                    </div>

                    <div class="row">
                        <div class="col-sm-12 col-md-12 text-center">
                            <div class="projects-header-quote">
                                <h3>Sharing my endeavors and passions...</h3>
                            </div>
                        </div>
                    </div>

                    <div class="row">
                        <div class="col-sm-4 col-md-4 text-center">
                            <div class="luvtalk-project">
                                <h3>LUV TALK Website</h3>
                                <p>Website, created during my first internship, which utilized HTML, CSS, PHP, Ajax, Javascript/jQuery, and Wordpress. </p>
                                <div class="row">
                                    <div class="col-sm-4 col-md-4 col-sm-offset-4 col-md-offset-4">
                                        <div class="project-more-info">
                                            <a class="btn btn-default btn-border" href="#">More</a>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>

                        <div class="col-sm-4 col-md-4 text-center">
                            <div class="personal-website-project">
                                <h3>Personal Website</h3>
                                <p>Enjoyable side project that was created to experiment with more HTML and CSS, but also provided an outlet to showcase my abilities and interests.</p>
                                <div class="row">
                                    <div class="col-sm-4 col-md-4 col-sm-offset-4 col-md-offset-4">
                                        <div class="project-more-info">
                                            <a class="btn btn-default btn-border" href="#">More</a>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>

                        <div class="col-sm-4 col-md-4 text-center">
                            <div class="strike-zone-project">
                                <h3>Strike Zone Analysis</h3>
                                <p>Data analytics project completed during my time at the Illinois Math and Science Academy which studied the baseball strike zone based on the state of the game.</p>
                                <div class="row">
                                    <div class="col-sm-4 col-md-4 col-sm-offset-4 col-md-offset-4">
                                        <div class="project-more-info">
                                            <a class="btn btn-default btn-border" href="#">More</a>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>

                    <div class="row">
                        <div class="col-sm-2 col-md-2 col-sm-offset-5 col-md-offset-5">
                            <div class="see-more-button">
                                <a class="btn btn-default btn-border" href="#">More Projects</a>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
