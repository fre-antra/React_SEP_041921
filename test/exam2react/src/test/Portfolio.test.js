import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Portfolio from '../components/Pages/Portfolio/Portfolio';

describe('Portfolio', () => {
    it('render without any errors', () => {
        const {container} = render(<Portfolio></Portfolio>);
        expect(container).toMatchSnapshot();
    })

    //it("has projects on it",  () => {
    //    const {container} = render(<Portfolio></Portfolio>);
    //    //console.log(container.findAllByText('Projects and Portfolio'));
    //    console.log(container);
    //})
})