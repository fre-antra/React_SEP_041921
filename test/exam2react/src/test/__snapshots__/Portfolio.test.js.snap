// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio render without any errors 1`] = `
<div>
  <div
    class="portfolio mb-60"
    data-testid="portfolio"
  >
    <div
      class="pt-60"
    />
    <div
      class="center-div undefined"
    >
      <p
        class="portfolio-section__intro"
        data-testid="portfolio-header"
      >
        Projects and Portfolio
      </p>
      <i
        class="portfolio-section__subintro mb-50"
      >
        Sharing my endeavors and passions...
      </i>
      <div
        class="mb-50"
      />
      <div
        class="ui container three column grid"
        data-testid="projects"
      >
        <div
          class="five wide computer sixteen wide mobile eight wide tablet column"
        >
          <div>
            <div
              class="item"
              style="background-image: url(luvtalklogo.png); width: 100%; background-position: center; background-size: cover;"
            >
              <div
                class="dark-layer item-layer"
                style="width: 100%;"
              >
                <div
                  class="center-div undefined"
                >
                  <p
                    class="portfolio-section__header"
                  >
                    LUV TALK Website
                  </p>
                  <div
                    class="mb-50"
                    style="width: 90%; text-align: center;"
                  >
                    <p>
                      Website, created during my first internship, which utilized HTML, CSS, PHP, Ajax, Javascript/jQuery, and Wordpress.
                    </p>
                  </div>
                  <button
                    class="btn__ghost-btn mb-50"
                  >
                    <a
                      href="#"
                    >
                      More
                    </a>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="five wide computer sixteen wide mobile eight wide tablet column"
        >
          <div>
            <div
              class="item"
              style="background-image: url(logo.svg); width: 100%; background-position: center; background-size: cover;"
            >
              <div
                class="dark-layer item-layer"
                style="width: 100%;"
              >
                <div
                  class="center-div undefined"
                >
                  <p
                    class="portfolio-section__header"
                  >
                    Personal Website
                  </p>
                  <div
                    class="mb-50"
                    style="width: 90%; text-align: center;"
                  >
                    <p>
                      Enjoyable side project that was created to experiment with more HTML and CSS, but also provided an outlet to showcase my abilities and interests.
                    </p>
                  </div>
                  <button
                    class="btn__ghost-btn mb-50"
                  >
                    <a
                      href="#"
                    >
                      More
                    </a>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="five wide computer sixteen wide mobile eight wide tablet column"
        >
          <div>
            <div
              class="item"
              style="background-image: url(mlbproject.png); width: 100%; background-position: center; background-size: cover;"
            >
              <div
                class="dark-layer item-layer"
                style="width: 100%;"
              >
                <div
                  class="center-div undefined"
                >
                  <p
                    class="portfolio-section__header"
                  >
                    Strike Zone Analysis
                  </p>
                  <div
                    class="mb-50"
                    style="width: 90%; text-align: center;"
                  >
                    <p>
                      Data analytics project completed during my time at the Illinois Math and Science Academy which studied the baseball strike zone based on the state of the game.
                    </p>
                  </div>
                  <button
                    class="btn__ghost-btn mb-50"
                  >
                    <a
                      href="#"
                    >
                      More
                    </a>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-60"
      />
      <button
        class="btn__ghost-btn "
      >
        <a
          href="#"
        >
          More Projects
        </a>
      </button>
      <div
        class="mb-50"
      />
    </div>
  </div>
</div>
`;
